segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -219,10 +239,11 @@ private void logEnd() {
 
         // Find the first chunk that contains the data.
         startOffsetForCurrentChunk.set(segmentMetadata.getFirstChunkStartOffset());
-        val readIndexTimer = new Timer();
+        boolean found = false;
```

Comment:
`found = (offset >= segmentMetadata.getLastChunkStartOffset()`

then update the `if` condition to use `found`